FlingEngine/Graphics/inc/GraphicsHelpers.h

```diff
@@ -2,88 +2,246 @@
+            VkCullModeFlags t_cullMode,
+            VkFrontFace t_frontFace,
+            VkPipelineRasterizationStateCreateFlags t_flags = 0
+        );
+
+        VkPipelineColorBlendStateCreateInfo PipelineColorBlendStateCreateInfo(
+            UINT32 t_attachmentCount,
+            const VkPipelineColorBlendAttachmentState* t_pAttachments
+        );
+
+        VkPipelineDepthStencilStateCreateInfo DepthStencilState(
+            VkBool32 t_depthTestEnable,
+            VkBool32 t_depthWriteEnable,
+            VkCompareOp t_depthCompareOp
+        );
+
+        VkPipelineViewportStateCreateInfo PipelineViewportStateCreateInfo(
+            UINT32 t_viewportCount,
+            UINT32 t_scissorCount,
+            VkPipelineViewportStateCreateFlags t_flags = 0
+        );
+
+        VkPipelineMultisampleStateCreateInfo PipelineMultiSampleStateCreateInfo(
+            VkSampleCountFlagBits t_rasterizationSamples,
+            VkPipelineMultisampleStateCreateFlags t_flags = 0
+        );
+
+        VkPipelineDynamicStateCreateInfo PipelineDynamicStateCreateInfo(
+            const std::vector<VkDynamicState>& t_pDynamicStates,
+            VkPipelineDynamicStateCreateFlags t_flags = 0
+        );
+
+        VkGraphicsPipelineCreateInfo PipelineCreateInfo(
+            VkPipelineLayout t_layout,
+            VkRenderPass t_renderPass,
+            VkPipelineCreateFlags t_flags = 0
+        );
+
+        VkVertexInputBindingDescription VertexInputBindingDescription(
+            UINT32 t_binding, 
+            UINT32 t_stride,
+            VkVertexInputRate t_inputRate
+        );
+
+        VkVertexInputAttributeDescription VertexInputAttributeDescription(
+            UINT32 t_binding,
+            UINT32 t_location,
+            VkFormat t_format,
+            UINT32 t_offset
+        );
+
+        VkViewport Viewport(
+            float t_width,
+            float t_height,
+            float t_minDepth,
+            float t_maxDepth
+        );
+
+        VkDescriptorSetLayoutBinding DescriptorSetLayoutBinding(
+            VkDescriptorType type,
+            VkShaderStageFlags stageFlags,
+            uint32_t binding,
+            uint32_t descriptorCount = 1);
+
+        VkWriteDescriptorSet WriteDescriptorSet(
+            VkDescriptorSet dstSet,
+            VkDescriptorType type,
+            uint32_t binding,
+            VkDescriptorBufferInfo* bufferInfo,
+            uint32_t descriptorCount = 1);
+    }
 
 }   // namespace Fling
 
 
-#define VK_CHECK_RESULT(f)																				\
-{																										\
-	VkResult res = (f);																					\
-	if (res != VK_SUCCESS)																				\
-	{																									\
-		F_LOG_FATAL("VkResult is {} in {} at line {}", res, __FILE__, __LINE__);						\
-		assert(res == VK_SUCCESS);																		\
-	}																									\
+#define VK_CHECK_RESULT(f)                                                                                \
```

Comment:
we really need to use the same Visual studio spacing settings :(

Reply:
we can resolve this next time we meet up